### PR TITLE
sys: move WAIT_FOR into time_units.h

### PR DESCRIPTION
**Pull request by fabiobaltieri**
WAIT_FOR uses k_us_to_cyc_ceil32, which is defined in time_units.h, so right now time_unitss.h and util.h include each others, potentially causing issues due to inconsistent include ordering.

Fix this by moving WAIT_FOR into time_units.h and dropping the time_units.h include from util.h.

### CODE REVIEW

File: include/zephyr/sys/util.h [hunk 1]
```diff
@@ -634,11 +634,6 @@ char *utf8_lcpy(char *dst, const char *src, size_t n);
 }
 #endif
 
-/* This file must be included at the end of the !_ASMLANGUAGE guard.
- * It depends on macros defined in this file above which cannot be forward declared.
- */
-#include <zephyr/sys/time_units.h>
-
```
Comment (marc-hb) [inline include/zephyr/sys/util.h:641]: This particular hunk is an exact revert of fbf851cdc4d8f0a so the commit message deserves a `Fixes: commit fbf851cdc4d8f0a`

**Review by marc-hb — DISMISSED** [review]
I see no problem with the code change but I tested this commit and it does NOT fix #62464 so please remove "Fixes #62464" from the commit message. You can still mention #62464 in the commit message to provide background information of course! Just don't say it fixes it because it makes no difference one way or the other.

File: include/zephyr/sys/util.h (hunk 1 above)
Comment (fabiobaltieri) [reply include/zephyr/sys/util.h:641]: Cool, done

**Comment by fabiobaltieri** [comment]
> I see no problem with the code change but I tested this commit and it does NOT fix #62464 so please remove "Fixes #62464" from the commit message. You can still mention #62464 in the commit message to provide background information of course! Just don't say it fixes it because it makes no difference one way or the other.

Cool, dropped the reference.

File: include/zephyr/sys/time_units.h [hunk 2]
```diff
@@ -1433,6 +1433,32 @@ static TIME_CONSTEXPR inline uint64_t k_ticks_to_cyc_ceil64(uint64_t t)
...
+ * request to read/write/initialize/reset, but useful for any expression.
+ *
+ * @param expr Truth expression upon which to poll, e.g.: XYZREG & XYZREG_EN
+ * @param timeout Timeout to wait for in microseconds, e.g.: 1000 (1ms)
+ * @param delay_stmt Delay statement to perform each poll iteration
+ *                   e.g.: NULL, k_yield(), k_msleep(1) or k_busy_wait(1)
+ *
+ * @retval expr As a boolean return, if false then it has timed out.
+ */
+#define WAIT_FOR(expr, timeout, delay_stmt)                                                        \
```
Comment (gmarull) [inline include/zephyr/sys/time_units.h:1451]: I'm not really happy with this movement, even though I understand the motivation/details. I wonder if time_units.h could be made independent of util.h by having a few internal macro copies from util.h... or, another solution would be to create a 3rd "common util" header (meant to be internal) that is included by util.h and time_units.h...

**Comment by fabiobaltieri** [comment]
Hey, lost momentum on this and it looks like it's not causing problems now and there's no obvious solution, I'll close this down, we can reopen if we have something more concrete to justifying putting more time into it.

**Pull request closed without merging**